File: GeneratedEntityReaderClasses/Components/DrugEffectComponent.cs
namespace GeneratedNoitaClasses;
public class DrugEffectComponent : NoitaComponentBase,  iNoitaType<DrugEffectComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public ConfigDrugFx drug_fx_target { get; set; } = new();
	public ConfigDrugFx m_drug_fx_current { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		drug_fx_target.Read(s);
		m_drug_fx_current.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		drug_fx_target.Write(s);
		m_drug_fx_current.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/DrugEffectComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from GeneratedClasses.ConfigDrugFx import ConfigDrugFx

from NoitaStream import NoitaStream
@dataclass
class DrugEffectComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    drug_fx_target:ConfigDrugFx = field(default_factory=ConfigDrugFx)
    m_drug_fx_current:ConfigDrugFx = field(default_factory=ConfigDrugFx)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.drug_fx_target.Read(s)
        self.m_drug_fx_current.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.drug_fx_target.Write(s)
        self.m_drug_fx_current.Write(s)

